pkgs/tools/misc/docker-slim/default.nix

```diff
@@ -0,0 +1,29 @@
+{ buildGoModule, fetchFromGitHub, stdenv, docker }:
+
+buildGoModule rec {
```

Comment:
please use `buildGoPackage`, because upstream vendorizes its dependencies

Reply:
Sure, thanks. Can you please tell me how to recognize that myself so I can be better in the future?

Reply:
> Sure, thanks. Can you please tell me how to recognize that myself so I can be better in the future?


If upstream has its dependencies in a `vendor/` directory, it's fine to use `buildGoPackage`, if `go.mod` is used, `buildGoModule` can be used :)

Reply:
sorry for the delay
>Can you please tell me how to recognize that myself so I can be better in the future?

The main difference between both, is that `buildGoModule` abuses fixed-output derivations which may not be preferable; because introduces impurities, such as network access https://github.com/NixOS/nix/issues/2270

As @Ma27 mentioned, we should `buildGoPackage` when a package vendorizes its dependencies (`vendor/` directory), since it's a regular expression and we don’t have to maintain a `deps.nix`
In the rest of the cases, the maintainer can choose which function to use